packages/core/src/modules/basic-messages/BasicMessagesApi.ts

```diff
@@ -91,7 +91,7 @@ export class BasicMessagesApi {
     await this.basicMessageService.deleteById(this.agentContext, basicMessageRecordId)
   }
 
-  private registerMessageHandlers(dispatcher: Dispatcher) {
-    dispatcher.registerMessageHandler(new BasicMessageHandler(this.basicMessageService))
+  private registerMessageHandlers(messageHandlerRegistry: MessageHandlerRegistry) {
+    messageHandlerRegistry.registerMessageHandler(new BasicMessageHandler(this.basicMessageService))
   }
```

Comment:
I'm thinking now, if we register these in the Api, wouldn't it mean they're registered every time a new agent is initialized? (in multi-tenancy). They share the same message handler registry